app/webpacker/components/Schedule/TimeZone.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { useMemo } from 'react';
+import { Dropdown } from 'semantic-ui-react';
+import i18n from '../../lib/i18n';
+
+const timeZoneOptions = Intl.supportedValuesOf('timeZone').map((timeZone) => ({
+  key: timeZone,
+  text: timeZone,
+  value: timeZone,
+}));
+
+export default function TimeZoneSelector({
+  venues,
+  activeTimeZone,
+  activeTimeZoneLocation,
+  dispatchTimeZone,
+}) {
+  const locationOptions = useMemo(
+    () => [
+      {
+        key: 'local',
+        text: i18n.t('competitions.schedule.timezone.local'),
+        value: 'local',
+      },
+      ...venues.map((venue, index) => ({
+        key: venue.name,
+        text: `${venue.name}'s`,
+        value: index,
+      })),
+      {
+        key: 'custom',
+        text: i18n.t('competitions.schedule.timezone.custom'),
+        value: 'custom',
+      },
+    ],
+    [venues],
+  );
+
+  return (
+    <div>
+      {i18n.t('competitions.schedule.timezone_setting')}
+      {' '}
+      <Dropdown
+        search
+        selection
+        value={activeTimeZoneLocation}
+        onChange={(_, data) => dispatchTimeZone({
+          type: 'update-location',
+          location: data.value,
+          venues,
+        })}
+        options={locationOptions}
+      />
+      {activeTimeZoneLocation === 'custom' && (
```

Comment:
I think this should always be visible. If I'm viewing venue A, and I select venue B's time zone, then I would like to know what time zone that is. It also avoids the unexpected disappearance of the second dropdown.

Maybe combining them into a single sentence? "You are viewing the calendar for the following location's time zone: < "your current location" | venue name | "custom location" > in < time zone >." Or something to that effect. Note the options  "your current location" and "custom location" replacing "your local" and "a custom".

If we prefer to keep them separate, then I would just change "The current custom timezone setting is" to "The calendar is currently being viewed in the time zone:". (Note "timezone" isn't a word, as far as I can tell; it should be time zone.)

Reply:
I still don't understand what the purpose behind these TZ selection dropdowns should actually be. And I don't mean "I don't understand" as a polite way of saying that you're wrong and I'm right. I genuinely don't understand.

Say I am selecting some subset of venues, and it shows the calendar view. The only potentially interesting use cases I can think of are:
- How does this schedule "interact" with other venues: I want to see the current schedule but in the time zone of other venues. So there should be one dropdown for selecting which venue's time zone to follow.
- Watching the live stream of a far away competition: I want to see how the schedule looks in relation to my local time zone. So I need one additional dropdown option for setting to my local (browser's) timezone
- Doing some custom calculation shenanigans, giving me the full freedom over selecting timezones.

In summary, I don't understand why you would ever want to have a dropdown offering you to follow a certain venue's TZ _and_ **at the same time** have a second dropdown allowing you to override the selection made by the venue dropdown.

Reply:
The use cases you list seem correct. (Plus you might want to plan to watch a live stream next weekend, but you're currently in a different time zone than you will be next weekend, so your local time zone would not be what you want. But that's probably not a common situation. Or you could see how jet lag might affect someone travelling from Germany to Worlds 2025, even if you live in a completely different time zone.)

Basically, WRT had some feedback on the schedule (there was an email thread many months ago), and we settled on the fact that it would be nice to be able to select any time zone. But just having a single (very long) drop-down of all time zones isn't nice. And just having a single drop-down of the venues (plus local) means you don't have all time zones. So I made 2 drop-downs which are synchronized (changing one automatically updates the other appropriately).

So I guess the short answer is that this is the nicest way I could find to allow the user to select any time zone, while still maintaining convenient access to venue time zones. If we don't need the user to be able to see arbitrary time zones then we could remove the custom option and the second drop-down.